### ISSUE TITLE
Use with dual-stack VPC causes cluster recreation

### ISSUE THREAD

**Issue by kliu47** [terraform-google-modules/terraform-google-kubernetes-engine#1863]
### TL;DR

When using a dualstack VPC, cluster tries to be re-created each run due to mismatch in enable_l4_ilb_subsetting

<img width="708" alt="image" src="https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/88392374/75c7d3bc-50ce-4e40-927d-cf359f1eed14">

### Expected behavior

Cluster would not be recreated

### Observed behavior

Terraform requires the cluster to be recreated

### Terraform Configuration

```hcl
module "gke" {
  source                          = "terraform-google-modules/kubernetes-engine/google//modules/beta-autopilot-private-cluster"
  version                         = "30.0.0"
  project_id                      = var.project_id
  name                            = "cluster-${var.environment}-test"
  region                          = var.region
  release_channel                 = "RAPID"
  network_project_id              = var.network_project
  network                         = var.network
  subnetwork                      = var.subnetwork
  ip_range_pods                   = var.ip_range_pods
  ip_range_services               = var.ip_range_svcs
  horizontal_pod_autoscaling      = true
  enable_vertical_pod_autoscaling = true
  enable_private_endpoint         = false
  enable_private_nodes            = true
  master_ipv4_cidr_block          = var.master_ipv4_cidr_block
  master_authorized_networks      = local.master_authorized_networks
  deletion_protection             = false
  gateway_api_channel             = "CHANNEL_STANDARD"
  network_tags = [
    "allow-gcp-load-balancer-health-check",
  ]
}
```

### Terraform Version

```sh
1.6.6
```

### Additional information

This issue might not be just limited to the module, but the core `google_container_cluster` itself

**Comment by Rycieos** [comment]
It is a module problem. This happens with the root module as well as the `beta-autopilot-private-cluster` and `beta-autopilot-public-cluster` modules, but not the `beta-public-cluster` module, as that can set `enable_l4_ilb_subsetting` with a variable.
